### ISSUE TITLE
Services depending on primary/secondary

### ISSUE THREAD

**Issue by jvinolas** [LINBIT/drbd-reactor#13]
It would be nice if services could be defined to run in the primary (as now does) and other services defined to be run in the secondary nodes. It's one thing we actually do in pacemaker.

**Comment by rck** [comment]
thank you for the feedback, can you elaborate a bit more on your use case?

our reasoning for `drbd-reactor::promoter` was that it should be simple and it should be closely tied to DRBD (which gives reactor free quorum and what we need for "cluster communication"). It runs services that need storage on DRBD. We intentionally did not design it as a full replacement for pacemaker. drbd-reactor is (intentionally) limited to what I described, if one needs a more generic cluster manager (which then comes with higher complexity (for the users and the ones maintaining the code)) one is better off with pacemaker. it is a fine cluster manager, nothing wrong with it, and nothing wrong with using the right tool for the job at hand.

**Comment by jvinolas** [comment]
Yeah, I've tried drbd-reactor and does perfectly what it says it will do!

We actually have our complex clústers with pacemaker and works well. But the only thing we would need to migrate hyperconvergent clústers to drbd-reactor (for simplicity) will be to allow setting OCF resources to be started on secondary nodes. We start an nfs server in the primary drbd node and then we mount it from secondary one's. It seems it would be quite easy to implement in drbd-reactor, don't know.

That would be nice for us, but I understand it's not planned in drbd-reactor and our use case is considered a pacemaker and we already have it set up with pacemaker.

Thanks!

**Comment by WanzenBug** [comment]
Not sure how feasible this is, but you may be able to create a mount unit for NFS with a `Conflicts=drbd-promote@<res>.service`.

Another alternative would be to handle this with the "user-mode-helper" plugin, listening for X->Secondary transitions of resources and then running the NFS mount afterwards.
